src/transformers/models/led/modeling_tf_led.py

```diff
@@ -1929,17 +1909,41 @@ def call(
                 attentions=all_self_attns,
             )
 
+    @tf.function
+    def compute_combined_attns_mask(self, inputs, input_shape, past_key_values_length):
```

Comment:
This function should be the same for all Bart-like models I think -> can we use the `# Copied from ...` statements here?

Reply:
Ok apparently it's not needed for Bart, which I don't get. `BartDecoder` and `LEDDecoder` compute the `combined_attns_mask` in exactly the same way. Why does Bart get away with:

```python
        # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
        if input_shape[-1] > 1:
            combined_attention_mask = _make_causal_mask(input_shape, past_key_values_length=past_key_values_length)
        else:
            combined_attention_mask = _expand_mask(
                tf.ones((input_shape[0], input_shape[1] + past_key_values_length)), tgt_len=input_shape[-1]
            )
        if inputs["attention_mask"] is not None and input_shape[-1] > 1:
            combined_attention_mask = combined_attention_mask + _expand_mask(
                inputs["attention_mask"], tgt_len=input_shape[-1]
            )
        if inputs["encoder_hidden_states"] is not None and inputs["encoder_attention_mask"] is not None:
            # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
            inputs["encoder_attention_mask"] = _expand_mask(inputs["encoder_attention_mask"], tgt_len=input_shape[-1])
```

and LED needs a `tf.function`?

Reply:
Actually in `master` for TFLED it's also:
`if inputs["attention_mask"] is None and and input_shape[-1] > 1:` -> see: https://github.com/huggingface/transformers/blob/248fa1ae72f01460c76c2450d747fa1cc29d85b0/src/transformers/models/led/modeling_tf_led.py#L1872 -> there might have been some merge conflicts...
Do you think we don't need the more difficult approach in LED?

Reply:
The code of TFLed actually looks exactly the same as for TFBart, I might have done a fix after your started your PR which probably led to some merge conflicts for TFLED... -> can we maybe try to align this functionality in TFLedDecoder and TFBartDecoder?

Reply:
Ok, let's try with a cleaner alignment then :)

Reply:
It works!! Just pushed the update :)

Reply:
Thanks for adapting it!